src/provision.ts

```diff
@@ -7,13 +7,15 @@ export const startProvisioning = async (button: LaunchButton) => {
   let device: BluetoothDevice | undefined;
   try {
     device = await navigator.bluetooth.requestDevice({
-      filters: [{ services: [IMPROV_BLE_SERVICE] }],
+      acceptAllDevices: true,
+      optionalServices: [IMPROV_BLE_SERVICE]
```

Comment:
This is not a good default setting and we should not merge it.

Do you have an example of a bad BLE implementation that is bad at broadcasting it's primary service? If it's a common problem we can consider adding a config option to make "acceptAll" an option.

Reply:
At the very least an "accept all" option would be great.

An example use case would be anytime I don't want improv to be my primary service, but a secondary service.
I am using nimble on a different project and there are several bugs related to primary services either getting cached, not working on chrome/safari, not broadcasting primary service consistently, or not advertising consistently.

Bluedroid doesn't seem to have an issue, it seems to be _very_ consistent in its advertisements.

Reply:
I can update this to allow an options hash for defaults, if that's the direction you'd like to go.

Reply:
Yes please make it opt-in to allow all by checking `button.hasAttribute('bluetooth-accept-all')`.

But you will also need to handle the case where a device is picked that does not have the required service.

Reply:
Can do! I'll see if i can get that over in the next few days!

Btw i appreciate these standards and this service, i cannot sing the praises of hassio/esphome/nabu casa enough. Fantastic companies and fantastic developers!